tests/src/Unit/OgAccessEntityTestBase.php

```diff
@@ -50,18 +52,39 @@ public function setup() {
     $entity_field_manager = $this->prophesize(EntityFieldManagerInterface::class);
     $entity_field_manager->getFieldDefinitions($entity_type_id, $bundle)->willReturn([$field_definition->reveal()]);
 
+    $group = $this->groupEntity()->reveal();
+    $group_type_id = $group->getEntityTypeId();
+
+    $storage = $this->prophesize(EntityStorageInterface::class);
+
     $entity_type_manager = $this->prophesize(EntityTypeManagerInterface::class);
     $entity_type_manager->getDefinition($entity_type_id)->willReturn($entity_type->reveal());
+    $entity_type_manager->getStorage($group_type_id)->willReturn($storage->reveal());
+
 
     $container = \Drupal::getContainer();
     $container->set('entity_type.manager', $entity_type_manager->reveal());
     $container->set('entity_field.manager', $entity_field_manager->reveal());
 
-    // Mock the results of Og::getEntityGroups().
+    // Mock the results of Og::getGroupIds().
     $r = new \ReflectionClass('Drupal\og\Og');
-    $reflection_property = $r->getProperty('entityGroupCache');
+    $reflection_property = $r->getProperty('cache');
     $reflection_property->setAccessible(TRUE);
-    $reflection_property->setValue(["$entity_type_id:$entity_id:1:" => [[$this->groupEntity()->reveal()]]]);
 
+    $identifier = [
+      'Drupal\og\Og::getGroupIds',
+      $entity_id,
+      NULL,
+      NULL,
+    ];
+
+    $identifier = implode(':', $identifier);
+
+    $group_ids = [$group_type_id => [$group->id()]];
+    $reflection_property->setValue([$identifier => $group_ids]);
+
+    // Mock the results of Og::getGroups().
+    $storage->loadMultiple(Argument::type('array'))->willReturn([$group]);
```

Comment:
I am very pleased with the Prophecy mocking framework. This was the first time I worked with it but it is very flexible and straightforward to understand. It takes a lot of the pain out of mocking.